### PR TITLE
Use matrix for Core GHA

### PR DESCRIPTION
**Pull request by paulb777**
Parallelize the `pod lib lint` platform runs with `matrix`

Inspired by #4752 by @wilhuff

#no-changelog

### CODE REVIEW

**Comment by paulb777** [comment]
cc: @ryanwilson @maksymmalyhin @mikehaney24

**Comment by ryanwilson** [comment]
Neat, great find!!

File: .github/workflows/core.yml [hunk 1]
```diff
@@ -10,15 +10,15 @@ on:
...
-  core:
+  pod_lib_lint:
     runs-on: macOS-latest
+
+    strategy:
+      matrix:
+        target: [ios, tvos, macos]
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
```
Comment (mikehaney24) [inline .github/workflows/core.yml]: why checkout@v2?

Comment (paulb777) [reply .github/workflows/core.yml]: See #4764 and https://github.com/actions/checkout

Comment (wilhuff) [reply .github/workflows/core.yml]: That PR doesn't really explain checkout@v2. There's some details in #4752 though, see specifically https://github.com/firebase/firebase-ios-sdk/pull/4752/commits/5ce0b2332cb6c4e39203689cda3e574af6aa39d4.

In short, the major difference between the versions is that v1 does a full clone including all branches, tags, and commits. checkout@v2 does a shallow clone of just one commit from the branch in question. This makes checkouts much faster.

**Pull request merged**